R/read_sc_all_alarms_telecare.R

```diff
@@ -18,15 +18,25 @@ read_sc_all_alarms_telecare <- function(sc_dvprod_connection = phs_db_connection
       "sending_location",
       "social_care_id",
       "period",
+      "period_start_date",
+      "period_end_date",
       "service_type",
       "service_start_date",
       "service_end_date"
     ) %>%
     dplyr::collect() %>%
-    # fix bad period (2017, 2020, 2021, and so on)
+    dplyr::distinct() %>%
+    dplyr::mutate(
+      period_start_date = dplyr::if_else(
+        .data$period == "2017",
+        lubridate::as_date("2018-01-01"),
+        .data$period_start_date
+      )
+    ) %>%
+    # fix bad period - 2017 only has Q4
     dplyr::mutate(
       period = dplyr::if_else(
-        grepl("\\d{4}$", .data$period),
+        .data$period == "2017",
         paste0(.data$period, "Q4"),
         .data$period
       )
```

Comment:
Looks fine, but just wondering how we deal with the other years in the incorrect format. e.g. 2020Q4 but is pulling through as 2020

Reply:
read your comment wrong, i see you said only Quarterly data is 2017. but dont know how to deal with the annual years.